### PR TITLE
Fix #26613 - Review summary block is not rendered in product page or product listing

### LINKED ISSUE #26613: Review summary block is not rendered in product page or product listing
### Preconditions (*)

1. Magento 2.3.3
2. An environment with multiple MySQL instances (3 in this case)

### Steps to reproduce (*)

1. Create a product that can be seen in a product list
2. Enable reviews
3. Add a review to the product

### Expected result (*)

1. Review summary block should display in the product cards and product page with the correct amount of stars

### Actual result (*)

1. Review summary block is not displayed at all

### Details
The root of the problem is in the function `\Magento\Review\Model\ReviewSummary::appendSummaryDataToObject`
Where the `$entityType` is set to 1 by default. In environments with multiple MySQL instances the autoincremental IDs are increased in intervals of 3. So we can't count on the product entity ID being 1, it might be 2 or 3. We fixed this issue by patching function `\Magento\Review\Block\Product\ReviewRenderer::getReviewsSummaryHtml` to fetch the product entity ID, instead of using the default 1

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Description (*)

Closes #26613 

### Fixed Issues (if relevant)

1. magento/magento2#26613: Review summary block is not rendered in product page or product listing

### Manual testing scenarios (*)

Please check fixed issue for testing steps

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by sdzhepa** [comment]
Put on hold due to https://github.com/magento/magento2/issues/26613#issuecomment-632091062

**Comment by Bartlomiejsz** [comment]
Hi @sdzhepa, since issue related was confirmed as reproduced, shouldn't this PR be un-holded?

**Comment by Bartlomiejsz** [comment]
@magento run all tests

**Comment by sidolov** [comment]
@magento run all tests

File: app/code/Magento/Review/Model/Review/SummaryData.php
Comment (sidolov) [inline app/code/Magento/Review/Model/Review/SummaryData.php]: Let's rename it to `AppendSummaryDataToObject ` with only one method `execute`

File: app/code/Magento/Review/Model/Review/SummaryData.php
Comment (sidolov) [inline app/code/Magento/Review/Model/Review/SummaryData.php]: Rename to  `execute`

File: app/code/Magento/Review/Block/Product/ReviewRenderer.php
Comment (sidolov) [inline app/code/Magento/Review/Block/Product/ReviewRenderer.php]: Please, make changes according to the BC development guide.

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by Bartlomiejsz** [comment]
@magento run all tests

File: app/code/Magento/Review/Block/Product/ReviewRenderer.php
Comment (Bartlomiejsz) [reply app/code/Magento/Review/Block/Product/ReviewRenderer.php]: Done

File: app/code/Magento/Review/Model/Review/SummaryData.php
Comment (Bartlomiejsz) [reply app/code/Magento/Review/Model/Review/SummaryData.php]: Done

File: app/code/Magento/Review/Model/Review/SummaryData.php
Comment (Bartlomiejsz) [reply app/code/Magento/Review/Model/Review/SummaryData.php]: Done

**Comment by Bartlomiejsz** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by Bartlomiejsz** [comment]
@magento run Functional Tests CE

**Comment by Bartlomiejsz** [comment]
@sidolov nice to see how core team handles community PRs, this one was opened almost one year ago, was under "review" since November, and today very similar changes are just pushed to codebase https://github.com/magento/magento2/commit/fb2f1a74ad2a9442c4c6ca39b85b42050a9f387b#diff-8c35e6e486082a9341a11e522aaef339c24b5383bade458b1f3ec263c27594d1
Maybe that would be worth to review PRs that are waiting at first, to not make people lose their time?

**Comment by sidolov** [comment]
Hi @Bartlomiejsz , sad to read that, sometimes the guys from the internal teams fixing the same issue but it's hard to find out that it's already fixed by the contributor in the public repository due to the large amount of open PRs. We will do our best to avoid such a situation in the future. I'm sorry for such  case.

**Pull request closed without merging**
